core/install/helm: Allow runner install to specify helm ref

demophoon opened the pull request:
This commit adds the `-k8s-helm-ref` flag to the runner install command to aid in testing helm changes without cutting a Waypoint-helm release by allowing Waypoint to configure which helm tarball to download by sha instead of by tag.

Fixes: #4199

demophoon committed:
runner/install/helm: Allow runner install to specify helm ref

This commit adds the `-k8s-helm-ref` flag to the runner install command
to aid in testing helm changes without cutting a Waypoint-helm release
by allowing Waypoint to configure which helm tarball to download by sha
instead of by tag.

paladin-devops commented on internal/installutil/k8s/k8s.go:
```diff
@@ -6,19 +6,21 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
```
Thank you for fixing the ordering here! 😄

demophoon committed:
Add changelog

demophoon committed:
docs: Add new flag to website

cicoyle reviewed (approved):
This is super helpful! :rocket:

demophoon committed:
core/install: Add `-k8s-helm-ref` to server installs

This commit replicates the `-k8s-helm-ref` flag from runner install to
server installations as well.

catsby reviewed (changes requested):
We've discussed this internally and are looking at alternatives. I recommend we close this for now

izaaklauer commented on internal/runnerinstall/k8s.go:
```diff
@@ -215,6 +219,13 @@ func (i *K8sRunnerInstaller) InstallFlags(set *flag.Set) {
+		Target: &i.Config.HelmRef,
+		Usage: "The git ref of the Helm chart to use for the Waypoint runner install. " +
+			"The ref may also be a branch name like 'main'. Takes precedent over -k8s-helm-version.",
+	})
```
If this is mostly for our internal use, i'd vote for making it a hidden flag so we don't noisify the docs.

```suggestion
Hidden: true,
	})
```

izaaklauer reviewed (changes requested):
Good idea!

ashleemboyer commented:
Hello @demophoon! 😊 This is a quick comment to let you know I'm updating this PR with main so it's up to date with the changes made in [#4277](https://github.com/hashicorp/waypoint/pull/4277).

Commit:
Pulling new link format workflows

Commit:
Revert "Pulling new link format workflows"

This reverts commit e5ba16fd6deeac5833ca81fb0644d273dc639ca9.

Commit:
Pulling new link format workflows from main

Commit:
Revert "Pulling new link format workflows from main"

This reverts commit ed5c07fabe0a03fc77b07f18d1629fcfb3459d51.

demophoon committed:
Update internal/runnerinstall/k8s.go

Co-authored-by: Izaak Lauer <8404559+izaaklauer@users.noreply.github.com>

briancain commented:
What's the status of this one @demophoon ?

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/waypoint?pullRequest=4202)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/waypoint?pullRequest=4202)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)


---
**1** out of **2** committers have signed the CLA.

- [x] demophoon
- [ ] Ashlee Boyer


---
**Ashlee Boyer** seems not to be a GitHub user.
You need a GitHub account to be able to sign the CLA. If you already have a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/waypoint?pullRequest=4202).</sub>
